tools/link.py

```diff
@@ -2764,6 +2764,8 @@ def process_dynamic_libs(dylibs, lib_dirs):
         exit_with_error(f'{os.path.normpath(dylib)}: shared library dependency not found: `{needed}`')
       to_process.append(path)
 
+    seen.add(os.path.basename(dylib))
```

Comment:
The `seen` set here is only used to prevent processing a library more than once.  It not used for anything else below.   So I don't see how this could mean that libraries were not being included.

Can you explain the bug you are trying to fix a litte more?  Do you have an example that fails perhaps?

Reply:
Yes. I opened an issue regarding this with the reproducer.

https://github.com/emscripten-core/emscripten/issues/21667

Reply:
My motivation of adding the dylib to seen is:

If the dylib does not have any needed libraries,
```
  for needed in dylink.needed:
```
there is no way for it to be added to the list of "seen".
Hence emcc does not know if a library, (eg. in the issue #21667
libbar.so) was visited already.

Reply:
"seen" will be reused in the while loop again. And since we are checking
`if needed in seen`, libbar will be discovered and the compilation will
continue.

Reply:
(at least for now)